devito/symbolics/manipulation.py

```diff
@@ -192,10 +193,10 @@ def run(expr):
                 rebuilt.append(expr.func(expr.lhs, ret, evaluate=False))
                 break
 
-    # Post-process the output
-    found = [Eq(v, k) for k, v in found.items()]
+        # Construct Eqs for temporaries
+        built.extend(expr.func(v, k) for k, v in list(found.items())[len(built):])
```

Comment:
`[len(built):]` ? 
isn't `built` empty before this line

Reply:
no idea what's going on here. Looks like a leftover from some changes I was trying. I will get back to you.

Good spot

Reply:
ah, no... 

this is inside a for-loop, so `built` gets populated at each iteration